lib/filestream.c

```diff
@@ -307,13 +307,13 @@ KTX_error_code ktxFileStream_getsize(ktxStream* str, ktx_size_t* size)
 
     assert(str->type == eStreamTypeFile);
 
-    // Need to flush so that fstat will return the current size.
-    // Can ignore return value. The only error that can happen is to tell you
-    // it was a NOP because the file is read only.
-#if (defined(_MSC_VER) && _MSC_VER < 1900) || defined(__MINGW32__)
-    // Bug in VS2013 msvcrt. fflush on FILE open for READ changes file offset
-    // to 4096.
-    if (str->data.file->_flag & _IOWRT)
+  // Need to flush so that fstat will return the current size.
+  // Can ignore return value. The only error that can happen is to tell you
+  // it was a NOP because the file is read only.
+#if (defined(_MSC_VER) && _MSC_VER < 1900) || defined(__MINGW32__) && !defined(_UCRT)
```

Comment:
Should we change `__MINGW32__` to `__MINGW64__`?

Reply:
Oh yeah I forgot to reply to that part of the issue. I don't think we really need to. It'll work the same either way but I can change it if you think it's a good idea.

Reply:
Please change it to `__MINGW64__`. Since KTX-Software only supports 64-bit builds, it will be less confusing.

Reply:
That was quick! Thanks.